modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala

```diff
@@ -80,10 +88,10 @@ class InstanceRoutes(instances: Instances[Future, Source[ByteString, Any], Sourc
       }
     }
 
-  protected def resourceRoutes: Route =
+  protected def resourceRoutes(cred: OAuth2BearerToken): Route =
```

Comment:
Using an `authenticator: AsyncAuthenticator[Caller]` with `authenticateOAuth2Async("*", authenticator)` directive we can provide an implicit caller that the `authorizeAsync` can make use of; i.e.:

```
private def authenticator(credentials: Credentials)(implicit ec: ExecutionContext): Future[Option[UserInfo]] =
    credentials match {
      case Credentials.Provided(token) =>
        verify(token)
          .map(ui => AuthenticatedCaller(token, ui))
          .recoverWith {
            case Unauthorized => Future.successful(None)
          }
      case Credentials.Missing =>
        Future.successful(Some(AnonymousCaller))
    }

private def hasPermission[A](resource: A, perm: Permission)(implicit caller: Caller): Future[Boolean] = ???

def route: Route = {
  authenticateOAuth2Async[Caller]("*", authenticator).apply { implicit caller =>
    authorizeAsync(hasPermission(schemaId, Read)) {
      complete("ok")
    }
  }
}
```
We could even have a compound directive based on authorizeAsync and hasPermission to simplify the call site.